nncf/quantization/algorithms/post_training/openvino_backend.py

```diff
@@ -0,0 +1,81 @@
+# limitations under the License.
+
+from pathlib import Path
+from typing import List
+
+import openvino.runtime as ov
+
+from nncf.common.graph.graph import NNCFNode
+from nncf.common.graph.transformations.commands import TargetType
+from nncf.openvino.graph.metatypes.openvino_metatypes import OVIfMetatype
+from nncf.openvino.graph.transformations.commands import OVExtractIfSubgraphCommand
+from nncf.openvino.graph.transformations.commands import OVOutputInsertionCommand
+from nncf.openvino.graph.transformations.commands import OVTargetPoint
+from nncf.openvino.graph.transformations.commands import OVUpdateIfSubgraphCommand
+from nncf.quantization.algorithms.post_training.backend import PostTrainingBackend
+
+
+class OVPostTrainingBackend(PostTrainingBackend):
+    IF_OP_MODEL_INPUT_PORTS = (0, 1)
+
+    @property
+    def if_node_metatype(self):
+        return OVIfMetatype
+
+    @staticmethod
+    def get_if_subgraph_input_names(model: ov.Model, if_node: NNCFNode, if_submodel_condition: bool) -> List[str]:
+        input_names = []
+        subgraph_port_id = 0 if if_submodel_condition else 1
+        name_to_node_mapping = {op.get_friendly_name(): op for op in model.get_ops()}
+        ov_node = name_to_node_mapping[if_node.node_name]
+        input_indices = [desc.input_index for desc in ov_node.get_input_descriptions(subgraph_port_id)]
+        input_names.extend([ov_node.input_values()[index].any_name for index in input_indices])
+        return input_names
+
+    @staticmethod
+    def get_if_cond_input_name(model: ov.Model, if_node: NNCFNode) -> str:
+        name_to_node_mapping = {op.get_friendly_name(): op for op in model.get_ops()}
+        ov_node = name_to_node_mapping[if_node.node_name]
+        return ov_node.input_values()[0].any_name
+
+    @staticmethod
+    def create_update_subgraph_command(
+        if_node: NNCFNode, if_submodel_condition: bool, subgraph_model: ov.Model
+    ) -> OVUpdateIfSubgraphCommand:
+        if_submodel_port_id = 0 if if_submodel_condition else 1
+        target_point = OVTargetPoint(TargetType.LAYER, if_node.node_name, if_submodel_port_id)
+        return OVUpdateIfSubgraphCommand(target_point, subgraph_model)
+
+    @staticmethod
+    def create_extract_if_subgraph_command(
+        if_node: NNCFNode, if_submodel_condition: bool
+    ) -> OVExtractIfSubgraphCommand:
+        return OVExtractIfSubgraphCommand(if_node, if_submodel_condition)
+
+    @staticmethod
+    def create_output_insertion_commands(model: ov.Model, if_node: NNCFNode) -> List[OVOutputInsertionCommand]:
+        commands = []
+        name_to_node_mapping = {op.get_friendly_name(): op for op in model.get_ops()}
+        ov_node = name_to_node_mapping[if_node.node_name]
+        for port_id in range(len(ov_node.inputs())):
+            commands.append(
+                OVOutputInsertionCommand(OVTargetPoint(TargetType.PRE_LAYER_OPERATION, if_node.node_name, port_id))
```

Comment:
It's not consistent that we call the `*_output_*` command that uses the `PRE_LAYER_*` approach.

Reply:
The word output here is inside the context of Model's output. The same function naming used in `nncf/openvino/statistics/aggregator.py` `_get_transformation_layout_extra_outputs`

Reply:
Let's name methods more clearly in the future even in case of the mistakes in the past.

Reply:
Please, add a suggestion if you have